ggml/src/ggml-metal.m

```diff
@@ -3468,10 +3428,46 @@ static void ggml_backend_metal_set_n_cb(ggml_backend_t backend, int n_cb) {
         }
     }
 
-    // TODO: setting encode_async here causes crash during the next ggml_metal_graph_compute call. why?
-    //ctx->encode_async = ^(size_t iter) {
-    //    ...
-    //};
+    ctx->encode_async = Block_copy(^(size_t iter) {
```

Comment:
In theory, we might want to call `ggml_backend_metal_set_n_cb()` multiple times in the future (although unlikely). So to make this more correct in this case, we should check if `ctx->encode_async == nil` here and if not, call `Block_release()`

Reply:
Ah, that had crossed my mind indeed, but didn't want to add more logic there than needed for now - worried about scope creep. I agree this makes total sense. Will add 👍